### PR TITLE
Downstream `node-template` pull

### PR DESCRIPTION
**Pull request by NukeManDan**
From diff of https://github.com/substrate-developer-hub/substrate-node-template/pull/246/files
- adds shell.nix 
- update's README to match
- ordering of `Cargo.toml`s for easy parsing/style/consistency

### CODE REVIEW

File: .maintain/node-template-release/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,616 @@
+# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
+version = 3
+
+[[package]]
+name = "adler"
```
Comment (shawntabrizi) [inline .maintain/node-template-release/Cargo.lock:6]: not sure about the creation of this new file?

**Review by shawntabrizi — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

File: bin/node-template/shell.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  pinned = builtins.fetchGit {
+    # Descriptive name to make the store path easier to identify
+    url = "https://github.com/nixos/nixpkgs/";
+    # Commit hash for nixos-unstable as of 2020-04-26
+    # `git ls-remote https://github.com/nixos/nixpkgs nixos-unstable`
+    ref = "refs/heads/nixos-unstable";
+    rev = "1fe6ed37fd9beb92afe90671c0c2a662a03463dd";
+  };
+  nixpkgs = import pinned { overlays = [ mozillaOverlay ]; };
+  toolchain = with nixpkgs; (rustChannelOf { date = "2021-07-06"; channel = "nightly"; });
```
Comment (bkchr) [inline bin/node-template/shell.nix:16]:
```suggestion
  toolchain = with nixpkgs; (rustChannelOf { date = "2021-09-14"; channel = "nightly"; });
```

**Comment by bkchr** [comment]
bot merge

File: .maintain/node-template-release/Cargo.lock (hunk 1 above)
Comment (NukeManDan) [reply .maintain/node-template-release/Cargo.lock:6]: Yeah, this was unintended. #9919

**Pull request merged**